src/harness/testcases/WINNF_FT_S_EPR_testcase.py

```diff
@@ -0,0 +1,364 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+import os
+import sas
+import sas_testcase
+from sas_test_harness import SasTestHarnessServer, generateCbsdRecords, \
+    generatePpaRecords
+from util import winnforum_testcase, configurable_testcase, writeConfig, \
+  loadConfig, makePpaAndPalRecordsConsistent
+
+
+class EscProtectionTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_EPR_1_default_config(self, filename):
+    """ Generates the WinnForum configuration for EPR.1. """
+
+    # Load ESC record
+    esc_record_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+
+    # Load devices info
+    device_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    # Moving device_1(Cat A) to a location within 40 meters of ESC sensor
+    device_1['installationParam']['latitude'] = 38.66815
+    device_1['installationParam']['longitude'] = -98.12986
+
+    device_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    # Moving device_2(Cat B) to a location within 80 meters of ESC sensor
+    device_2['installationParam']['latitude'] = 38.66782
+    device_2['installationParam']['longitude'] = -98.13033
+
+    device_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    # Moving device_3(Cat A) to a location outisde 40 meters of ESC sensor
+    device_3['installationParam']['latitude'] = 38.6679
+    device_3['installationParam']['longitude'] = -98.13017
+
+    device_4 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_d.json')))
+    # Moving device_4(Cat B) to a location outside 80 meters of ESC sensor
+    device_4['installationParam']['latitude'] = 38.66747
+    device_4['installationParam']['longitude'] = -98.12986
+
+    # Load Grant requests
+    grant_request_1 = json.load(
```

Comment:
Since this is checking both in-band and out-of-band CBSDs, a comment here would make it clearer?

Reply:
As suggested, added comments for grant requests at appropriate places